server/data_common/data_adaptor.py

```diff
@@ -83,6 +83,10 @@ def query_var_array(self, term_var):
     def query_obs_array(self, term_var):
         pass
 
+    @abstractmethod
+    def colors(self):
```

Comment:
I believe the style in this class is `get_foo`, not `foo`, so should be `get_colors()`